packages/tailwindcss-transformer/src/index.ts

```diff
@@ -73,12 +73,37 @@ export function transform(code: string, ctx: { styleCache: StyleCache }) {
       }
       this.traverse(path);
     },
-    // visit cn function calls containing TW classes
+    // visit a `className` property within any object containing TW classes
+    visitObjectProperty(path) {
+      const node = path.node;
+      if (path.node.key.type === 'Identifier' && path.node.key.name === 'className') {
+        visitNode(node, ctx);
+      }
+      this.traverse(path);
+    },
+    // visit function calls containing TW classes
     visitCallExpression(path) {
       const node = path.node;
-      if (node.callee.type === 'Identifier' && node.callee.name === 'cn') {
+      // `className` concatenation functions
+      if (node.callee.type === 'Identifier' && ['cn', 'cx', 'clsx'].includes(node.callee.name)) {
         visitNode(node, ctx);
       }
+      // cva functions (note: only compatible with cva@1.0)
+      if (
+        node.callee.type === 'Identifier' &&
+        node.callee.name === 'cva' &&
+        node.arguments[0].type === 'ObjectExpression'
```

Comment:
Although in practice you wouldn't call `cva` without arguments, I'm thinking to be safe we should use optional chaining here to prevent a crash if you do `cva()`:

```suggestion
        node.arguments[0]?.type === 'ObjectExpression'
```